test/common/config/delta_subscription_impl_test.cc

```diff
@@ -1,165 +1,86 @@
-// "unsubscribe from [resource name I have never before referred to]" requests.
-TEST_F(DeltaSubscriptionImplTest, AddThenRemove) {
+TEST_F(DeltaSubscriptionImplTest, UpdateResourcesCausesRequest) {
   startSubscription({"name1", "name2", "name3"});
-  subscription_->pause(); // Pause because we're testing multiple updates in between request sends.
+  expectSendMessage({"name4"}, {"name1", "name2"}, Grpc::Status::GrpcStatus::Ok, "", {});
+  subscription_->updateResources({"name3", "name4"});
+  expectSendMessage({"name1", "name2"}, {}, Grpc::Status::GrpcStatus::Ok, "", {});
   subscription_->updateResources({"name1", "name2", "name3", "name4"});
-  subscription_->updateResources({"name1", "name2", "name3"});
-  InSequence s;
-  expectSendMessage({}, {"name4"}, Grpc::Status::GrpcStatus::Ok, "", {});
-  subscription_->resume();
+  expectSendMessage({}, {"name1", "name2"}, Grpc::Status::GrpcStatus::Ok, "", {});
+  subscription_->updateResources({"name3", "name4"});
+  expectSendMessage({"name1", "name2"}, {}, Grpc::Status::GrpcStatus::Ok, "", {});
+  subscription_->updateResources({"name1", "name2", "name3", "name4"});
+  expectSendMessage({}, {"name1", "name2", "name3"}, Grpc::Status::GrpcStatus::Ok, "", {});
+  subscription_->updateResources({"name4"});
 }
 
-// add/remove/add == add.
-TEST_F(DeltaSubscriptionImplTest, AddRemoveAdd) {
+TEST_F(DeltaSubscriptionImplTest, PauseBlocksRequest) {
   startSubscription({"name1", "name2", "name3"});
   subscription_->pause();
+
+  expectSendMessage({"name4"}, {"name1", "name2"}, Grpc::Status::GrpcStatus::Ok, "", {});
+  // If not for the pause, these updates would make the expectSendMessage fail due to too many
+  // messages being sent.
+  subscription_->updateResources({"name3", "name4"});
   subscription_->updateResources({"name1", "name2", "name3", "name4"});
-  subscription_->updateResources({"name1", "name2", "name3"});
+  subscription_->updateResources({"name3", "name4"});
   subscription_->updateResources({"name1", "name2", "name3", "name4"});
-  InSequence s;
-  expectSendMessage({"name4"}, {}, Grpc::Status::GrpcStatus::Ok, "", {});
-  subscription_->resume();
-}
+  subscription_->updateResources({"name3", "name4"});
 
-// remove/add/remove == remove.
-TEST_F(DeltaSubscriptionImplTest, RemoveAddRemove) {
-  startSubscription({"name1", "name2", "name3"});
-  subscription_->pause();
-  subscription_->updateResources({"name1", "name2"});
-  subscription_->updateResources({"name1", "name2", "name3"});
-  subscription_->updateResources({"name1", "name2"});
-  InSequence s;
-  expectSendMessage({}, {"name3"}, Grpc::Status::GrpcStatus::Ok, "", {});
   subscription_->resume();
 }
 
-// Starts with 1,2,3. 4 is added/removed/added. In those same updates, 1,2,3 are
-// removed/added/removed. End result should be 4 added and 1,2,3 removed.
-TEST_F(DeltaSubscriptionImplTest, BothAddAndRemove) {
+TEST_F(DeltaSubscriptionImplTest, PauseQueuesAcks) {
```

Comment:
Since you're doing cleanup, could you add comments above `TEST_F`s where they are non-trivial (e.g. here) and running commentary at least at the block level below? Basically what you've done below for `DeltaSubscriptionStateTest` (really great!)

Reply:
Done. Also, while looking at the ACK queue test, added a simpler just-a-single-ACK test.